app/javascript/controllers/conversation_message_controller.js

```diff
@@ -0,0 +1,15 @@
+"use strict"
+
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static values = {
+    scrollIntoView: { type: Boolean, default: false }
+  }
+
+  connect() {
+    if (this.scrollIntoViewValue) {
+      this.element.scrollIntoView(true, { behavior: "smooth" })
+    }
```

Comment:
The way I've implemented this, the page doesn't scroll to the bottom when it first loads. That's probably the wrong way to go about it, but I don't think I really like it when I open a webpage and it jumps straight to the bottom. I left it as-is because it feels like there's more to think about.

Reply:
@robacarp Hmm, that’s notable. Now that you mention it, i’m browsing around chatgpt and as I click to different conversations in the sidebar, they don’t load the full thing and then scroll me all the way down. Frankly, some of them are really long! Somehow, as the page loads you’re already at the bottom of the page and can scroll up.

I tried a bunch of long conversations where I scrolled up really quickly to see if they were doing progressive loading as I scroll back in time and I was never able to catch it doing that.

But maybe we’re thinking about this wrong? Maybe we need to reverse the flex stack direction, reverse the scroll direction, and just implement an old-school infinite scroll which loads progressively, but just in reverse?

Reply:
@krschacht I was trivially able to get this to work as expected. It's snappy too, at least in localhost.